File: frontend/src/types/coaching.ts
export interface ToolCall {
  name: string;
  description: string;
  arguments?: Record<string, unknown>;
}

export interface ChatMessage {
  role: "assistant" | "user";
  content: string;
  timestamp: string;
  tool_calls?: ToolCall[];  // Tool calls made by AI (for UI display)
}

export interface ProposedChange {
  type: string;
  description: string;
  details: Record<string, unknown>;
  accepted: boolean | null;
}

export interface HabitPerformance {
  habit_id: string;
  title: string;
  completed_count: number;
  total_days: number;
  rate: number;
}

export interface TrackerTrend {
  tracker_id: string;
  name: string;
  values: number[];
  trend: string;
}

export interface PerformanceSnapshot {
  period_start: string;
  period_end: string;
  habits: HabitPerformance[];
  tracker_trends: TrackerTrend[];
}

export interface SessionSummary {
  key_points: string[];
  habits_added: string[];
  next_check_in: string | null;
  action_items: string[];
}

export interface CoachingSession {
  id: string;
  goal_id: string;
  user_id: string;
  trigger: string;
  status: string;
  performance_snapshot: PerformanceSnapshot | null;
  messages: ChatMessage[];
  proposed_changes: ProposedChange[];
  summary: SessionSummary | null;
  created_at: string;
  resolved_at: string | null;
}

File: backend/app/models/coaching.py
from datetime import datetime
from typing import Optional, Any

from pydantic import BaseModel


class ChatMessage(BaseModel):
    role: str  # "assistant" | "user"
    content: str
    timestamp: datetime


class HabitPerformance(BaseModel):
    habit_id: str
    title: str
    completed_count: int
    total_days: int
    rate: float


class TrackerTrend(BaseModel):
    tracker_id: str
    name: str
    values: list[float] = []
    trend: str = "stable"


class PerformanceSnapshot(BaseModel):
    period_start: str
    period_end: str
    habits: list[HabitPerformance] = []
    tracker_trends: list[TrackerTrend] = []


class SessionSummary(BaseModel):
    """Summary of a coaching session - like 'minutes of meeting'"""
    key_points: list[str] = []  # Main discussion points
    habits_added: list[str] = []  # Habits created/modified
    next_check_in: Optional[str] = None  # When to reconnect (e.g., "in 7 days")
    action_items: list[str] = []  # Things user should do


class CoachingSession(BaseModel):
    id: str
    goal_id: str
    user_id: str = "default"
    trigger: str = "scheduled_review"
    status: str = "active"
    performance_snapshot: Optional[PerformanceSnapshot] = None
    messages: list[ChatMessage] = []
    summary: Optional[SessionSummary] = None  # Summary when session ends
    created_at: datetime
    resolved_at: Optional[datetime] = None
